scenarios/kubernetes_bazel.py

```diff
@@ -52,15 +56,58 @@ def get_version():
 
 def main(args):
     """Trigger a bazel build/test run, and upload results."""
+    # pylint:disable=too-many-branches
+    if args.install:
+        for install in args.install:
+            if not os.path.isfile(install):
+                raise ValueError('Invalid install path: %s' % install)
+            check('pip', 'install', '-r', install)
+
     check('bazel', 'clean', '--expunge')
     res = 0
     try:
-        if args.build:
+        if args.build is not None:
             check('bazel', 'build', *args.build.split(' '))
         if args.release:
             check('bazel', 'build', *args.release.split(' '))
         if args.test:
-            check('bazel', 'test', *args.test.split(' '))
+            if args.test == 'affected':
```

Comment:
maybe rather than using a special test name here, add another flag?

the value of `args.test` could then instead be used in the bazel query to restrict the targets searched, e.g.
`kind(.*_binary, rdeps(set(%s), set(%s)))' % (args.test, packages)`